docs/chat_message_history.ipynb

```diff
@@ -1,79 +1,184 @@
-                "\n",
-                "Save chat messages into `DATABASE`."
-            ]
-        },
-        {
-            "cell_type": "markdown",
-            "metadata": {},
-            "source": [
-                "## Pre-reqs"
-            ]
-        },
-        {
-            "cell_type": "code",
-            "execution_count": null,
-            "metadata": {
-                "tags": []
-            },
-            "outputs": [],
-            "source": [
-                "%pip install PACKAGE_NAME"
-            ]
-        },
-        {
-            "cell_type": "code",
-            "execution_count": 3,
-            "metadata": {
-                "tags": []
-            },
-            "outputs": [],
-            "source": [
-                "from PACKAGE import LOADER"
-            ]
-        },
-        {
-            "cell_type": "markdown",
-            "metadata": {},
-            "source": [
-                "## Basic Usage"
-            ]
-        },
-        {
-            "cell_type": "code",
-            "execution_count": null,
-            "metadata": {},
-            "outputs": [],
-            "source": []
-        }
-    ],
-    "metadata": {
-        "kernelspec": {
-            "display_name": "Python 3 (ipykernel)",
-            "language": "python",
-            "name": "python3"
-        },
-        "language_info": {
-            "codemirror_mode": {
-                "name": "ipython",
-                "version": 3
-            },
-            "file_extension": ".py",
-            "mimetype": "text/x-python",
-            "name": "python",
-            "nbconvert_exporter": "python",
-            "pygments_lexer": "ipython3",
-            "version": "3.10.6"
-        }
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6-0_o3DxsFGi"
+      },
+      "source": [
+        "Google Database\n",
+        "\n",
+        "Use [Google Memorystore for Redis](https://cloud.google.com/memorystore/docs/redis/memorystore-for-redis-overview) to store chat message history for LangChain."
+      ]
     },
-    "nbformat": 4,
-    "nbformat_minor": 4
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dWakBoPnsFGj"
+      },
+      "source": [
+        "## Pre-reqs"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "EudfLv_UsFGk"
+      },
+      "source": [
+        "### Setting Up a Memorystore for Redis Instance\n",
+        "\n",
+        "Before proceeding, an active Memorystore for Redis instance is needed to store chat message history:\n",
+        "\n",
+        "* Create a Memorystore for Redis Instance (>= 5.0): If an instance doesn't exist, follow the instructions at https://cloud.google.com/memorystore/docs/redis/create-instance-console to create a new one. Ensure the version is greater than or equal to 5.0.\n",
+        "* Obtain Endpoint: Note the endpoint associated with the instance."
```

Comment:
Is there a cloud.google.com doc page that we can link with more guidance?

Reply:
Yes, it is shown in L33: "* Create a Memorystore for Redis Instance (>= 5.0): If an instance doesn't exist, follow the instructions at https://cloud.google.com/memorystore/docs/redis/create-instance-console to create a new one. Ensure the version is greater than or equal to 5.0.\n",